core/templates/dev/head/pages/community-dashboard-page/services/translate-text.service.spec.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests that the translate-text service is working as
+ * expected.
+ */
+
+require(
+  'pages/community-dashboard-page/services/' +
+  'translate-text.service.ts');
+
+describe('TranslateTextService', function() {
+  var TranslateTextService, $httpBackend;
+  var mockTranslateBackendObject;
+  var callbackObject;
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.inject(function($injector) {
+    $httpBackend = $injector.get('$httpBackend');
+    TranslateTextService = $injector.get('TranslateTextService');
+    mockTranslateBackendObject = {
```

Comment:
rename this to `mockStateWiseTextForTranslation` or `mockStateWiseTranslatableText`. (I would prefer the 1st one)